natss/pkg/dispatcher/dispatcher.go

```diff
@@ -360,11 +361,11 @@ func (s *SubscriptionsSupervisor) ProcessChannels(ctx context.Context, chanList
 	return nil
 }
 
-func (s *SubscriptionsSupervisor) getChannelReferenceFromHost(host string) (eventingchannels.ChannelReference, error) {
+func (s *SubscriptionsSupervisor) getChannelReferenceFromHost(host url.URL) (eventingchannels.ChannelReference, error) {
 	chMap := s.getHostToChannelMap()
-	cr, ok := chMap[host]
+	cr, ok := chMap[host.Path]
 	if !ok {
-		return cr, fmt.Errorf("Invalid HostName:%q. HostName not found in any of the watched natss channels", host)
+		return cr, fmt.Errorf("Invalid HostName:%v. HostName not found in any of the watched natss channels", host)
```

Comment:
Why not preserving the quotes? It makes it easier to distinguish the input from the message IMO.

Reply:
+1